src/minishell.c

```diff
@@ -37,9 +37,30 @@ static void	sighandler(int signal)
 
 int	readcmd(char *cmd)
 {
-	getcmd(cmd, MAX_COMMAND_LEN);
-	cmd[ft_strcspn(cmd, "\n")] = '\0';
-	return (ft_strlen(cmd));
+	char	temp_cmd[MAX_COMMAND_LEN];
+	int		total_len;
+	int		len;
+
+	total_len = 0;
+	while (1)
+	{
+		getcmd(temp_cmd, MAX_COMMAND_LEN);
+		len = (int)ft_strcspn(temp_cmd, "\n");
+		if (len > 0 && temp_cmd[len - 1] == '\\')
+		{
+			temp_cmd[len - 1] = ' ';
+			ft_strncpy(cmd + total_len, temp_cmd, len);
+			total_len += len;
+		}
+		else
+		{
+			ft_strncpy(cmd + total_len, temp_cmd, len + 1);
+			total_len += len + 1;
+			break;
+		}
```

Comment:
I tested the code and it works very well, even though it does not work perfectly, still impressive.
But I have doubt that there might be a better or safer or more straight or more intuitive way to concatenate command(cmd), maybe using ft_strlcpy() with ft_strlcat().
Let's find out better code and better implementation later!

Reply:
I'm sure there's a better way, but as always, one of the problems is norm :)... I'm not sure if I'm going to go to 42 or not today, but if I do, I'll let you know what the maximum value on the command line is.
what I learned from the last project, sometimes we need to design function features that fit or have the potential to fit into the norm. I will make some changes today and resubmit.